### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by K0ppai**
Hello reviewers,
I've worked on the following in this pull request,

1. Create a class `Classroom`
2. Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`
3. Create a class `Book`
4. Create a class `Rental`
5. Create the `many-to-many` relationship between `Person` and `Book` using the intermediate class `Rental`

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (ridwanediallo) [inline book.rb]: - [ ] Excellent job using the attr accessor to set the `title`, `author`, and `rentals.` Amazing work was also done with the initialize method 👏💪. So, I kindly propose that you include a method called `add rental` in the `Book` class to ensure that a user can rent a book. 👌😊
An illustration of how to use the `add rental` method to implement the `has many` relationship between a `book` and a `rental` is provided below. :point_down: 

```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require './nameable'
 
 class Person < Nameable
```
Comment (ridwanediallo) [inline person.rb:3]: - [ ] Amazing work was also done with the initialize method 👏💪. So to ensure that a person can rent a book, I kindly propose that you include a method called `add rental` in the `Book` class. 👌😊
An illustration of how to use the `add rental` method to implement the `has many` relationship between a `person` and a `rental` is provided below. :point_down: 

```
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (ridwanediallo) [inline rental.rb]: - [ ] Please make sure that the `person` and `book` are set with the `attr accessor` property. If not, you will only be able to read the values of a `person` and a `book` from a "Rental" instance; you will not be able to use assignment to directly alter these values. For instance, you could use `rental.book` to retrieve the associated book object, but you couldn't use `rental.book = new book` to replace the rental's associated book. 👌😊

```suggestion
   attr_accessor :date, :book, :person
```

**Review by ridwanediallo — CHANGES_REQUESTED** [review]
# Changes Required ♻️
## Hi @K0ppai , 
AlmostThereReachGIF

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Most of the project requirements have been met with success. ✔


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 3 above)
Comment (K0ppai) [reply rental.rb]: @ridwanediallo, Thank you for the detailed explanation. I appreciate it.😁

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @K0ppai👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with associations. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by K0ppai** [comment]
Thanks @ridwanediallo, @thecodechaser for the reviews. 😄

**Pull request closed without merging**